Integrations/integration-MaxMind_GeoIP2.yml

```diff
@@ -0,0 +1,285 @@
+description: Enriches IP addresses
+detaileddescription: 'The MaxMind GeoIP2 integration allows you to query the MaxMind
+  API service and retrieve a JSON of all details. '
+configuration:
+- display: API Key
+  name: apikey
+  defaultvalue: ""
+  type: 0
+  required: true
+- display: Account ID
+  name: account
+  defaultvalue: ""
+  type: 0
+  required: true
+- display: Use system proxy
+  name: proxy
+  defaultvalue: ""
+  type: 8
+  required: false
+- display: Trust any certificate (unsecure)
+  name: insecure
+  defaultvalue: ""
+  type: 8
+  required: false
+- display: Service Version
+  name: mode
+  defaultvalue: city
+  type: 15
+  required: true
+  options:
+  - country
+  - city
+  - insights
+script:
+  script: |
+    import requests
+    import collections
+    from requests.auth import HTTPBasicAuth
+    # disable insecure warnings
+    requests.packages.urllib3.disable_warnings()
+
+
+
+
+    '''GLOBAL VARS'''
+    BASE_URL = 'https://geoip.maxmind.com/geoip/v2.1/'
+    APIKEY = demisto.params().get('apikey')
+    ACCOUNT_ID = demisto.params().get('account')
+    MODE = demisto.params().get('mode')
+    INSECURE = demisto.params().get('insecure')
+    PROXY = demisto.params().get('proxy')
+    if not demisto.params().get('proxy', False):
+        del os.environ['HTTP_PROXY']
+        del os.environ['HTTPS_PROXY']
+        del os.environ['http_proxy']
+        del os.environ['https_proxy']
+
+    hr_headers = [
+        'IP',
+        'Domain',
+        'ASN',
+        'Organization',
+        'ISP',
+        'Location',
+        'Accuracy Radius',
+        'User Type',
+        'Continent',
+        'ISO Code',
+        'Country',
+        'Registered Country',
+        'TimeZone',
+        'City',
+        'Subdivision',
+        'Is TOR Exit Node',
+        'Is Hosting Provider',
+        'Is Anonymous']
+
+
+
+
+    '''HELPER FUNCTIONS'''
+    def http_request(method, query, MODE):
+        if method is 'GET':
+            headers = {
+                'Content-Type': 'application/json',
+                'Accept': 'application/json'
+            }
+        r = requests.request(
+            method,
+            BASE_URL + MODE +'/'+ query,
+            headers=headers,
+            verify=INSECURE,
+            auth=HTTPBasicAuth(ACCOUNT_ID, APIKEY)
+        )
+        if r.status_code is not 200:
+            return_error('Error in API call to MaxMind [%d] - %s' % (r.status_code, r.reason))
+        return r
+
+
+    # Allows nested keys to be accesible
+    def makehash():
+        return collections.defaultdict(makehash)
+
+
+    def format_results(raw):
```

Comment:
All the fields in the response are optional??

Reply:
Yeah, there are three different service levels which return different amounts of data depending on the service. Of the data returned, it may or may not contain everything that's available.